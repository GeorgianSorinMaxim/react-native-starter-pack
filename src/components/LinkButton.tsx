import * as React from "react";
import { StyleSheet, Text, View } from "react-native";
import Icon from 'react-native-vector-icons/FontAwesome';
import { RectButton } from "react-native-gesture-handler";

type LinkButtonProps = {
  label: string,
  onPress: () => void,
  isLastOption: boolean
}

const LinkButton = ({ label, onPress, isLastOption }: LinkButtonProps) => {
  return (
    <RectButton
      style={[styles.option, isLastOption && styles.lastOption]}
      onPress={onPress}
    >
      <View style={styles.buttonContainer}>
        <View style={styles.iconContainer}>
          <Icon name="link" size={22} color="rgba(0,0,0,0.35)" />
        </View>
        <Text style={styles.optionText}>{label}</Text>
      </View>
    </RectButton>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: "row"
  },
  iconContainer: {
    marginRight: 12
  },
  option: {
    backgroundColor: "#fdfdfd",
    paddingHorizontal: 15,
    paddingVertical: 15,
    borderWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: 0,
    borderColor: "#ededed"
  },
  lastOption: {
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  optionText: {
    fontSize: 15,
    alignSelf: "flex-start",
    marginTop: 1
  }
});

export default LinkButton;