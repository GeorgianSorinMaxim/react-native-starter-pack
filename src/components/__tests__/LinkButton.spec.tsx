import * as React from "react";
import renderer from "react-test-renderer";

import LinkButton from "../LinkButton";

it(`renders correctly`, () => {
  const defaultProps = {
    label: 'Nandos',
    onPress: () => {},
    isLastOption: false
  };
  const tree = renderer.create(<LinkButton {...defaultProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
