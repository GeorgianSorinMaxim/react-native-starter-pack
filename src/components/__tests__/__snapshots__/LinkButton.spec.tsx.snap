// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.105}
  collapsable={false}
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Array [
        Object {
          "backgroundColor": "#fdfdfd",
          "borderBottomWidth": 0,
          "borderColor": "#ededed",
          "borderWidth": 0.5,
          "paddingHorizontal": 15,
          "paddingVertical": 15,
        },
        false,
      ],
    ]
  }
  underlayColor="black"
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 12,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,0.35)",
              "fontSize": 22,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÉÅ
      </Text>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 15,
          "marginTop": 1,
        }
      }
    >
      Nandos
    </Text>
  </View>
</RNGestureHandlerButton>
`;
